File: python/3rd_week/Inheritance/Manusia.py
class Manusia:
    # implementasi class manusia

    # constructor
    def __init__(self):
        self.nama = ""
        self.alamat = ""
        self.no_ktp = ""
        self.no_telepon = ""

    # set method
    def setNama(self, nama):
        self.nama = nama
    
    def setAlamat(self, alamat):
        self.alamat = alamat
    
    def setNoKtp(self, no_ktp):
        self.no_ktp = no_ktp
    
    def setNoTelepon(self, no_telepon):
        self.no_telepon = no_telepon
    
    # get method
    def getNama(self):
        return self.nama
    
    def getAlamat(self):
        return self.alamat
    
    def getNoKtp(self):
        return self.no_ktp
    
    def getNoTelepon(self):
        return self.no_telepon
    

File: cpp/3rd_Week/MariMencoba3/Manusia.cpp
#include <iostream>
#include <string>

using namespace std;

class Manusia
{
    private:
        /* data */
        string nama, alamat, noKtp, Telepon;
    public:
        Manusia() {}
        string getNama() {
            return nama;
        }

        void setNama(string nama) {
            this->nama = nama;
        }

        string getAlamat() {
            return alamat;
        }

        void setAlamat(string alamat) {
            this->alamat = alamat;
        }

        string getNoKtp() {
            return noKtp;
        }

        void setNoKtp(string noKtp) {
            this->noKtp = noKtp;
        }

        string getTelepon() {
            return Telepon;
        }

        void setTelepon(string telepon) {
            this->Telepon = telepon;
        }

        ~Manusia() {}
};

class Arsitek: public Manusia{
    private:
        string kodeArsitek, kantor, tahunBekerja;

//    constructor
    public:
        Arsitek(){}

    //    set dan get method
            //set method
        void setKodeArsitek(string kodeArsitek) {
            this->kodeArsitek = kodeArsitek;
        }

        void setKantor(string kantor) {
            this->kantor = kantor;
        }

        void setTahunBekerja(string tahunBekerja) {
            this->tahunBekerja = tahunBekerja;
        }

            //get method
        string getKodeArsitek() {
            return kodeArsitek;
        }

        string getKantor() {
            return kantor;
        }

        string getTahunBekerja() {
            return tahunBekerja;
        }

        ~Arsitek(){};
};

class PenjualBahanBangunan: public Manusia {
    private:
        string namaToko, alamatToko;

//    constructor
    public:
        PenjualBahanBangunan(){}

    //    set and get method
        string getNamaToko() {
            return namaToko;
        }

        void setNamaToko(string namaToko) {
            this->namaToko = namaToko;
        }

        string getAlamatToko(){
            return alamatToko;
        }

        void setAlamatToko(string alamatToko){
            this->alamatToko = alamatToko;
        }

        ~PenjualBahanBangunan(){};
};